pkg/defaultRules/defaultRules.yaml

```diff
@@ -1891,6 +1891,8 @@ rules:
             not:
               enum:
                 - Kustomization
+                - ClusterRole
```

Comment:
This means `ClusterRole` can omit the `name` field altogether, I think this should be separated into 2 cases:
1. enforce the `name` field for all relevant resources
2. enforce the `name` field format only for some of those resources